src/target/riscv/riscv-013.c

```diff
@@ -3556,7 +3620,7 @@ static int read_memory_abstract(struct target *target, target_addr_t address,
 			result = write_abstract_arg(target, 1, address + c * size, riscv_xlen(target));
 			if (result != ERROR_OK) {
 				LOG_TARGET_ERROR(target, "Failed to write arg1.");
-				return result;
+				return MEM_ACCESS_STATUS_FAILED_ABSTRACT_ARG_WRITE_FAILED;
```

Comment:
There could be just one generic return status `MEM_ACCESS_FAILED_DM_ACCESS_ERROR` that would cover all cases when access to the DM registers failed.

Reply:
Addressed